File: src/main/java/me/sangbo/simple/Send.java
package me.sangbo.simple;

import com.rabbitmq.client.Channel;
import com.rabbitmq.client.Connection;
import me.sangbo.ConnectionUtils;

import java.io.IOException;
import java.util.concurrent.TimeoutException;

/**
 * Created by 桑博 on 2018/6/12.
 */
public class Send {


  public static final String QUEUE_NAME= "queue_sangbo";

  public static void main(String[] args) throws IOException, TimeoutException {


    Connection connection = ConnectionUtils.getConnection();
    Channel channel = connection.createChannel();
    channel.queueDeclare(QUEUE_NAME,false,false,false,null);
    String msg = "桑博，今天晚上去吃饭吗！饿了。。。";

    channel.basicPublish("",QUEUE_NAME,null,msg.getBytes());
    channel.close();
    connection.close();

  }

}

File: src/main/java/me/sangbo/roulting/Send.kt
package me.sangbo.roulting

import me.sangbo.ConnectionUtils
import java.nio.charset.Charset

/**
 * Created by 桑博 on 2018/6/14.
 */
private const val EXCHANGE_NAME = "exchange_sangbo_direct"
private const val ROULTING_KEY = "info";

fun main(args: Array<String>) {

    val connection = ConnectionUtils.getConnection();
    val channel = connection.createChannel();
    channel.exchangeDeclare(EXCHANGE_NAME,"direct");
    val msg = "测试";
    channel.basicPublish(EXCHANGE_NAME,ROULTING_KEY,null,msg.toByteArray());
    channel.close();
    connection.close();

}